pkg/resource/deploy/step.go

```diff
@@ -155,7 +155,10 @@ func (s *SameStep) Apply() (resource.Status, StepCompleteFunc, error) {
 	// We can only do this if the provider is actually a same, not a skipped create.
 	if providers.IsProviderType(s.new.Type) && !s.skippedCreate {
 		if s.Deployment() != nil {
-			err := s.Deployment().SameProvider(s.new)
+			// we use the _old_ state here because we want to ensure that the provider is registered under the
```

Comment:
This is missing a bit of the sentence it seems.  Also we're using only the old *inputs*, not the whole old state.

Reply:
Oops yup, let me rewrite that properly.